src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -3356,6 +3356,26 @@ public void testOomScoreAdj() throws Exception {
   }
 
   @Test
+  public void testAutoRemove() throws Exception {
+    requireDockerApiVersionAtLeast("1.25", "AutoRemove");
+
+    // Pull image
+    sut.pull(BUSYBOX_LATEST);
+
+    final ContainerConfig config = ContainerConfig.builder()
+        .image(BUSYBOX_LATEST)
+        .hostConfig(HostConfig.builder()
+                        .autoRemove(true) // Default is false
+                        .build())
+        .build();
+
+    final ContainerCreation container = sut.createContainer(config, randomName());
+    final ContainerInfo info = sut.inspectContainer(container.id());
+
+    assertThat(info.hostConfig().autoRemove(), is(true));
```

Comment:
In this test, you are just setting the option on container creation and verifying that it was indeed set correctly. A better test would be to start and stop the container, verifying that it does get removed.